src/module/system/damage/weapon.ts

```diff
@@ -512,6 +512,12 @@ class WeaponDamagePF2e {
             ignoredResistances,
         };
 
+        // If a weapon deals no base damage, remove all bonuses, penalties, and modifiers to it.
```

Comment:
remove all basic dice and modifiers? Its not quite removing all.

Reply:
it = base damage

Reply:
I see, I interpret it as "weapon". Ambiguous but sure.